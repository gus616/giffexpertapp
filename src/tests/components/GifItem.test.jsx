import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react';
import GifItem from '../../components/GifItem';


describe('<GifItem/> tests', ()=>{
    test('should match the snapshot', ()=> {
        const image = {title: 'vinland', url: 'https://media4.giphy.com/media/uRb9QFo9ZeMes/giphy.gif?cid=9ca20bfbfyjodd4oplg6io7oi0vj1gqzspmu8yyhx9a01xas&rid=giphy.gif&ct=g'};
        const gifItemComponent = renderer.create(<GifItem {...image}/>); 
        const {container} = render(<GifItem {...image}/>); //with 'render' you get access to all the 
        expect(gifItemComponent).toMatchSnapshot();
        expect(container).toMatchSnapshot();
    });

    test('must show the image with the url and alt indicated', ()=> {
        const image = {title: 'vinland', url: 'https://media4.giphy.com/media/uRb9QFo9ZeMes/giphy.gif?cid=9ca20bfbfyjodd4oplg6io7oi0vj1gqzspmu8yyhx9a01xas&rid=giphy.gif&ct=g'};
        render(<GifItem {...image}/>);
        //screen.debug();
        const {src , alt} = screen.getByRole('img');
        expect( src ).toBe( image.url );
        expect( alt ).toBe( image.title );
    });

    test('must show the title in the component', ()=> {
        let title = 'Saitama';
        let url = 'https://media4.giphy.com/media/11OlEyIaGC6Jsk/giphy.gif?cid=9ca20bfb9qh4kbdpjjdid6pa22a35po551y74uvrfdh0i0gn&rid=giphy.gif&ct=g';
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText( title )).toBeTruthy();
    });
});

