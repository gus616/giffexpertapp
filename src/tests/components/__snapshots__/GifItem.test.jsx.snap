// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem/> tests should match the snapshot 1`] = `
<div
  className="card"
>
  <img
    alt="vinland"
    src="https://media4.giphy.com/media/uRb9QFo9ZeMes/giphy.gif?cid=9ca20bfbfyjodd4oplg6io7oi0vj1gqzspmu8yyhx9a01xas&rid=giphy.gif&ct=g"
  />
  <p>
    vinland
  </p>
</div>
`;

exports[`<GifItem/> tests should match the snapshot 2`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="vinland"
      src="https://media4.giphy.com/media/uRb9QFo9ZeMes/giphy.gif?cid=9ca20bfbfyjodd4oplg6io7oi0vj1gqzspmu8yyhx9a01xas&rid=giphy.gif&ct=g"
    />
    <p>
      vinland
    </p>
  </div>
</div>
`;
